drivers/wifi/wfx200/wfx200.c

```diff
@@ -0,0 +1,525 @@
+/* SiLabs WFX200 WiFi Module with SPI
+ * or SDIO Interface
+ *
+ * Copyright (c) 2021 grandcentrix GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT silabs_wfx200
+
+#define LOG_MODULE_NAME wifi_wfx200
+#define LOG_LEVEL CONFIG_WIFI_LOG_LEVEL
+
+#include <sl_wfx.h>
+#undef BIT
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <string.h>
+#include <errno.h>
+
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/gpio.h>
+#include <drivers/spi.h>
+#include <net/net_pkt.h>
+#include <net/net_if.h>
+#include <net/ethernet.h>
+
+#include "wfx200_internal.h"
+
+struct wfx200_dev wfx200_0 = { 0 };
+
+static void wfx200_iface_init(struct net_if *iface)
+{
+	const struct device *dev = net_if_get_device(iface);
+	struct wfx200_dev *context = dev->data;
+
+	LOG_DBG("Interface init");
+	context->iface = iface;
+	net_if_set_link_addr(iface, context->sl_context.mac_addr_0.octet,
+			     sizeof(context->sl_context.mac_addr_0.octet),
+			     NET_LINK_ETHERNET);
+	ethernet_init(iface);
+	context->state = WFX200_STATE_INTERFACE_INITIALIZED;
+	net_if_flag_set(iface, NET_IF_NO_AUTO_START);
+}
+
+int wfx200_send(const struct device *dev, struct net_pkt *pkt)
+{
+	struct wfx200_dev *context = dev->data;
+	size_t len = net_pkt_get_len(pkt), frame_len;
+	sl_wfx_send_frame_req_t *tx_buffer;
+	sl_status_t result;
+	int status;
+
+	switch (context->state) {
+	case WFX200_STATE_AP_MODE:
+		if (!(context->sl_context.state & SL_WFX_AP_INTERFACE_UP)) {
+			return -EIO;
+		}
+		break;
+	case WFX200_STATE_STA_MODE:
+		if (!(context->sl_context.state & SL_WFX_STA_INTERFACE_CONNECTED)) {
+			return -EIO;
+		}
+		break;
+	default:
+		return -ENODEV;
+	}
+	frame_len = SL_WFX_ROUND_UP(len, 2);
+	result = sl_wfx_allocate_command_buffer((sl_wfx_generic_message_t **)&tx_buffer,
+						SL_WFX_SEND_FRAME_REQ_ID,
+						SL_WFX_TX_FRAME_BUFFER,
+						frame_len + sizeof(sl_wfx_send_frame_req_t));
+	if (result != SL_STATUS_OK) {
+		return -ENOMEM;
+	}
+	if (net_pkt_read(pkt, tx_buffer->body.packet_data, len)) {
+		status = -EIO;
+		goto error;
+	}
+	if (context->state == WFX200_STATE_AP_MODE) {
+		result = sl_wfx_send_ethernet_frame(tx_buffer,
```

Comment:
You could also set a variable to either `SL_WFX_SOFTAP_INTERFACE` or `SL_WFX_STA_INTERFACE` and only call `sl_wfx_send_ethernet_frame()` once (no copy-and-paste).

Reply:
i did that now :heavy_check_mark: